Add assigner for crawl tasks [ZZ-45]

batanoffs opened the pull request:
Added constant `CRAWL_TASK_STATUS` to store the status for the tasks, please give any suggestions if it should be placed else where . Check line 13 to 20;

batanoffs committed:
add assigner for crawl tasks

DanielDimitrov5 commented on src/assigner.ts:
```diff
@@ -1,14 +1,94 @@
+import prisma from "./db/prisma/prisma";
+import logger from "./utils/logger";
+
+// Declare and define constant for crawl task status.
+// Better to move it but where to? Suggestion db/prisma/constants - file constants.ts
+const CRAWL_TASK_STATUS = {
+    PENDING: "PENDING",
+    IN_PROGRESS: "IN_PROGRESS",
+    COMPLETED: "COMPLETED",
+    FAILED: "FAILED",
+};
```
Import this 'import { CrawlTaskStatus } from "@prisma/client"'
This is the enum type from the schema. From there you can access the statuses directly.
No need for this.

DanielDimitrov5 commented on src/assigner.ts:
```diff
@@ -1,14 +1,94 @@
+            // Check if crawl task is created for current source
+            const getTask = await prisma.crawlTasks.findFirst({
+                where: {
+                    source: {
```
This table is called 'Sources'.
Did you even test it?

DanielDimitrov5 commented on src/assigner.ts:
```diff
@@ -1,14 +1,94 @@
+
+                // Check if status is NOT pending or in-progress and create new one
+                if (
+                    getTask.status !== CRAWL_TASK_STATUS.PENDING ||
+                    getTask.status !== CRAWL_TASK_STATUS.IN_PROGRESS
```
Check this logic once again.

DanielDimitrov5 reviewed (changes requested):
Test the assigner when the issues are fixed.
When you start, it should create just 3 tasks and nothing more.

batanoffs committed:
update logger settings to work with num 1g

batanoffs committed:
update assigner logic and fix issues, wrap in try/catch to handle errors

batanoffs commented:
Changes:
- Update logic for assigner as requested. 
- Added try/catch. 
- Updated logger to work with numbers instead of bool

During tests:
- my code didn't generate new warnings
- tested and created 3 tasks, when when status is changed to "COMPLETE", it generated a new one

For review - at line 37:
```js 
const getTask = await prisma.crawlTasks.findFirst({
       where: {
                 status: CrawlTaskStatus.PENDING || CrawlTaskStatus.IN_PROGRESS,
                 source_id: source.id,
               },
       });
```
Should i use findMany instead of findFirst?

DanielDimitrov5 reviewed (approved):
Perfect one!
